Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityInputManager.cs

```diff
@@ -97,64 +97,61 @@ private void InitializeInternal()
             bool addedComponents = false;
 
 #if UNITY_EDITOR
-
-            var eventSystems = UnityEngine.Object.FindObjectsOfType<EventSystem>();
-            var standaloneInputModules = UnityEngine.Object.FindObjectsOfType<StandaloneInputModule>();
-
-            CameraCache.Main.transform.position = Vector3.zero;
-            CameraCache.Main.transform.rotation = Quaternion.identity;
-
-            if (eventSystems.Length == 0)
+            if (!UnityEditor.EditorApplication.isPlaying)
             {
-                FocusProvider.UIRaycastCamera.gameObject.EnsureComponent<EventSystem>();
-                addedComponents = true;
-            }
-            else
-            {
-                bool raiseWarning;
+                var eventSystems = UnityEngine.Object.FindObjectsOfType<EventSystem>();
+                var standaloneInputModules = UnityEngine.Object.FindObjectsOfType<StandaloneInputModule>();
 
-                if (eventSystems.Length == 1)
+                CameraCache.Main.transform.position = Vector3.zero;
+                CameraCache.Main.transform.rotation = Quaternion.identity;
+
+                if (eventSystems.Length == 0)
                 {
-                    raiseWarning = eventSystems[0].gameObject != focusProvider.gameObject;
+                    FocusProvider.UIRaycastCamera.gameObject.EnsureComponent<EventSystem>();
+                    addedComponents = true;
                 }
                 else
                 {
-                    raiseWarning = true;
-                }
+                    bool raiseWarning;
 
-                if (raiseWarning)
-                {
-                    Debug.LogWarning("Found an existing event system in your scene. The Mixed Reality Input System requires only one, and must be found on the focus provider.");
-                }
-            }
+                    if (eventSystems.Length == 1)
+                    {
+                        raiseWarning = eventSystems[0].gameObject != focusProvider.gameObject;
```

Comment:
Why does the event system need to be on the same GameObject as the `FocusProvider`? None of the existing example scenes are set up this way.

Reply:
Because the existing example scenes were setup incorrectly.

It needs to be there because the FocusProvider depends on it.

Reply:
I think the event system is added to the UIRaycastCamera that is found under the camera.

I'll update the debug log to reflect that fact.

Reply:
Mainly this is so we don't have to call `FindObjectOfType<T>` anywhere. We keep track of the event system and input module.

Reply:
> I think the event system is added to the UIRaycastCamera that is found under the camera.

Right, but we're checking against a different GameObject here. Should this be `focusProvider.UIRaycastCamera.gameObject`, to match the `FocusProvider.UIRaycastCamera.gameObject.EnsureComponent<EventSystem>()` line above? Otherwise, we're adding to one GameObject and checking against a different one.